main.go

```diff
@@ -1224,13 +1228,128 @@ func populatePackages(currentPackage string, onlyUpdates bool, onlyLatest bool,
 		}
+
 		packageList = append(packageList, packageWrapper)
 	}
 
 	return packageList, nil
 }
 
-// func generateChanges(genpatch bool, save bool, commit bool, onlyUpdates bool, print bool) {
+// downloadIcons should only be used in a local machine by manual execution.
+// It will download all icons that contain URLs from the index.yaml file, if it is already downloaded it will keep it.
+// All downloaded icons will be saved in the assets/logos directory and for all of them will be created a upstream.yaml file in the respective packages folder.
+// The upstream.yaml file will contain the icon field under ChartMetadata to be overriden in the index.yaml file later by the overrideIcons function.
+func downloadIcons(c *cli.Context) {
+	currentPackage := os.Getenv(packageEnvVariable)
+	icons.CheckFilesStructure() // stop execution if file structure is not correct
+
+	packageList, err := populatePackages(currentPackage, false, false, false)
+	if err != nil {
+		logrus.Fatal(err)
+	}
+
+	// Convert packageList to PackageIconMap
+	var entriesPathsAndIconsMap icons.PackageIconMap = make(icons.PackageIconMap)
+	for _, pkg := range packageList {
+		entriesPathsAndIconsMap[pkg.Name] = icons.PackageIconOverride{
+			Name: pkg.Name,
+			Path: pkg.Path,
+			Icon: pkg.LatestStored.Metadata.Icon,
+		}
+	}
+
+	// Download all logos or retrieve the ones already downloaded
+	downloadedIcons := icons.DownloadFiles(entriesPathsAndIconsMap)
+
+	logrus.Infof("Finished downloading and saving upstream yml files with icons for override")
+	logrus.Infof("Downloaded %d icons", len(downloadedIcons))
+}
+
+// overrideIcons will get the package list and override the icon field in the index.yaml file with the downloaded icons.
+// It will also test if the icons are correctly overridden and if the index.yaml file is correctly written.
+// If the test fails, it will return an error and the user should check the logs for more information.
+// It will only work if the downloadIcons function was previously executed at some point in time.
+// The function will not download the icons, it will only override the icon field in the index.yaml file.
+// Before overriding the icons, it will check if the necessary conditions are met at parsePackageListToPackageIconList() function, if not it will skip the package.
+func overrideIcons() {
+	currentPackage := os.Getenv(packageEnvVariable)
+	iconOverride := true
+	icons.CheckFilesStructure() // stop execution if file structure is not correct
+
+	// populate all possible packages
+	packageList, err := populatePackages(currentPackage, false, false, false)
+	if err != nil {
+		logrus.Fatal(err)
+	}
+
+	// parse only the packages that have the necessary conditions for icon override
+	packageIconList := parsePackageListToPackageIconList(packageList)
+
+	err = overwriteIndexIconsAndTestChanges(packageIconList)
+	if err != nil {
+		logrus.Errorf("Failed to overwrite index icons: %v", err)
+	}
+
+	err = commitChanges(packageList, iconOverride)
```

Comment:
Do we want to make it optional to commit the changes? I think we want to call `overrideIcons()` in `prepareCharts()` and `stageChanges()`, but we don't want to commit in those cases.

To be honest, I wish that `partner-charts-ci` never did anything with `git` at all. I wish it just modified files, and then left it up to the user to stage and commit changes. It would be a lot simpler that way. But that conversation is for another time.

Reply:
IMO, one of the strengths of `partner-charts-ci` is the automatic commits, we don't leave room for human-made errors and keep consistency in the workflow which can be more automated later. 

About `prepareCharts()` and `stageChanges()` we don't want to call `overrideIcons` there, `overrideIcons` is for resolving the problem in the short term so we don't have to change already released `charts and assets`. 

We want to generate the proper `chart, assets, and index.yaml` of newer versions with `prepare` + `patch` + whatever it is for the rest of the process.